analyzer.py

```diff
@@ -17,17 +17,17 @@ def is_build_failed(job):
 
 def was_built_in_last_24h(job):
     if 'lastBuild' in job:
-        build_date_time = datetime.utcfromtimestamp(job['lastBuild']['timestamp'] / 1e3) # to proper timestamp
-        time_diff_in_hours = (datetime.now() - build_date_time).total_seconds() / 60 / 60 # seconds to hours
+        build_date_time = datetime.utcfromtimestamp(job['lastBuild']['timestamp'] / 1e3)  # to proper timestamp
+        time_diff_in_hours = (datetime.now() - build_date_time).total_seconds() / 60 / 60  # seconds to hours
         if time_diff_in_hours < 24:
             return True
     return False
 
 
-def filter_jobs(all_jobs):
+def filter_jobs(all_jobs, only_built_in_last_24_hours=True):
```

Comment:
this functions seems to bit messy now... conditions are not really easy to read imho

since we have now additional moving part (getting all failed jobs or just those from last 24h) i would split it into 2 separate functions, e.g
`def filter_failed_jobs(all_jobs)` - it should return only failing jobs, no matter when they have been built
`def filter_build_in_last_24h_jobs(all_jobs)` - it should return jobs build in last 24h, no matter job result

it will be a bit code duplication, but now functions have clear and single responsibility and expose nice interface - you could use them as building blocks, e.g
`all_failed = filter_failed_jobs(all_jobs)`
`newest_failed = filter_failed_jobs(filter_build_in_last_24h_jobs(all_jobs))`

Reply:
Isn't that what parameters are for? Otherwise we will have two almost exact copies of the same function. I personally find this rather errorprone, because when you have to make changes to that function you have to remember to do it twice. If someone who is not familiar with the code does a change now, they might not notice that they have to do the change twice. 
```
def filter_jobs(all_jobs):
    failed_jobs = []
    for job in all_jobs:
        if is_build_failed(job):
                    failed_jobs.append(job)
    return failed_jobs

def filter_built_in_last_24h_jobs(all_jobs):
    failed_jobs = []
    for job in all_jobs:
        if is_build_failed(job) and was_built_in_last_24h(job):
                    failed_jobs.append(job)
    return failed_jobs
```

Reply:
following up on slack